### PR TITLE
[Tokens Table] Sort by 24h volume

### LINKED ISSUE #210: Sort Tokens table by 24h volume instead of Total volume
**Description**
Currently the Tokens table on the Homepage is sorted by Total volume, the plan is to change that (if possible) to be sorted by 24h volume.

Also we need to add a visual indicator on the `24h volume` column header to make it clear we are sorting by that column in descending order.

image

### PR DESCRIPTION
**Pull request by matextrem**
# Summary

Closes #210 

- Refactor code to get tokens table info in just one query to the subgraph.
- Sort the tokens table by 24h Volume instead of Total Volume.

Screen Shot 2022-09-14 at 00 38 18

# To Test

1.  Open the `home` page and wait for the tokens table to load.
*  You'll see the tokens table is now sorted by 24h volume.

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @matextrem , cool changes!
Happy to see the arrow that currently shows a default sorting on the table!

However, the arrow is shifted to the 2ns line when the table is displayed on a reduced screen. Can we fix it somehow?
another line

Also, the arrow looks weird (from my perspective) in a mobile view. It is displayed on each card, and it is not clear that it shows a default sorting
do not show on mobile
Maybe we should remove it from the mobile view and just type something like "Sorted by Volume (24h)' under the search field?
image

@alongoni , @matextrem , WDYT?

Thanks!

File: src/hooks/useGetTokens.ts [hunk 1]
```diff
@@ -79,6 +27,7 @@ export function useGetTokens(networkId: Network | undefined, tableState: TableSt
     )?.averagePrice
 
     return {
+      lastDayUsdTimestamp: lastDayTimestampFrom,
```
Comment (ramirotw) [inline src/hooks/useGetTokens.ts:30]: what's this `lastDayUsdTimestamp` value? is that the Last Updated timestamp value? If so, shouldn't this be picked from the `timestamp` value response?

**Review by ramirotw — CHANGES_REQUESTED** [review]
Looking good!

A few notes:

1. Check how the `Price (24h)` and `Price (7d)` values are calculated, we have that data through the hourlyTotals field so it shouldn't be empty
image

2. The last updated timestamp isn't too clear. Could you move the timestamp to the next line and add also minute precision: `Last Updated: 2022-09-13 21:30h`
image

3. The averagePrice that is picked in:

```js
    const priceUsd = data.tokenHourlyTotals[0]?.averagePrice
```

is now wrong as the `tokenHourlyTotals` array is sorted in ascending order and you are now picking the last week price and not the current one.

4. is the `addHistoricalData` function still needed? I added that because we were using 2 different queries to build the table, but now we can do just 1 query for everything, couldn't we just use the `processTokenData` or similar?

**Comment by matextrem** [comment]
@elena-zh regarding the first comment, I cannot find a solution for this. Maybe @alongoni can come out with an idea. 
@ramirotw I checked the way the values are calculated and it seems to be fine (I checked it in https://explorer.cow.fi/ and some values are undefined as well). IDK if we are doing something wrong with the query though.

**Comment by alfetopito** [comment]
Did not check the code yet, but one thing that confuses me:
Screen Shot 2022-09-15 at 10 49 16

We are on the 15th, but the top token was last updated on the 11th?
Maybe this is more of a Graph question, but the data does not look accurate

File: src/hooks/useGetTokens.ts [hunk 2]
```diff
@@ -95,33 +43,43 @@ export function useGetTokens(networkId: Network | undefined, tableState: TableSt
...
+        const response = await COW_SDK.cowSubgraphApi.runQuery<{ tokenDailyTotals: TokenDailyTotalsResponse[] }>(
+          GET_TOKENS_QUERY,
+          {
+            lastDayTimestamp,
+            lastWeekTimestamp,
+          },
+          { chainId: network },
+        )
+        if (response) {
+          const tokensData = {} as { [tokenId: string]: TokenData }
```
Comment (alfetopito) [inline src/hooks/useGetTokens.ts:62]: What about:

```suggestion
          const tokensData: { [tokenId: string]: TokenData } = {}
```

**Review by alfetopito — COMMENTED** [review]
Just a small nitpick

**Comment by elena-zh** [comment]
Hey @matextrem , I think that the easiest way to fix the issue with the arrow is to remove a space between the label and the arrow
image

My bad, I did not mentioned it in the previous comment, but it would be great to add a way of sorting to the mobile view: Sorted by Volume(24h): from highest to lowest (or so). WDYT?

As for values that are displayed in Tokens tables in Dev and in the current PR, they appear to be different. I suppose that they should be the same
image

**Comment by elena-zh** [comment]
Hey @matextrem , 
The arrow LGTM. 
Tokens last updated date looks great, however, I'd not start 'Updated' with the capital letter. 
image

The same is for 'From highest to lowest": it would be great not to start From with the capital letter. 

As for amounts and numbers in the table. they still look different in comparison with the Dev app version. 
ыешдд ыуу вуааукутсув
And now there is a difference in 24h volumes. Frankly, I can't find how to check the daily token volume in the graph,       maybe you could help me with a request to check what number is a correct one. 
Anyways, I think that numbers should match...

Thanks

**Comment by elena-zh** [comment]
Hey @matextrem , changes look better! 
I can see matching amounts in Price(24h) column now. 
But still  see that amounts are differ for Price (7d) and Volume (24h) columns image

Could you please take a look at it? 

Thanks

**Comment by ramirotw** [comment]
> And now there is a difference in 24h volumes. Frankly, I can't find how to check the daily token volume in the graph, maybe you could help me with a request to check what number is a correct one.
> Anyways, I think that numbers should match...

@elena-zh To test the values you can run [this](https://api.thegraph.com/subgraphs/name/cowprotocol/cow/graphql?query=query+%28%24address%3A+String%2C+%24from%3A+Int%2C+%24to%3A+Int%29+%7B%0A%09tokenHourlyTotals%28%0A%09%09orderBy%3A+timestamp%0A%09%09orderDirection%3A+desc%0A%09%09where%3A+%7B+token%3A+%24address%2C+timestamp_gte%3A+%24from%2C+timestamp_lt%3A+%24to+%7D%0A%09%29+%7B%0A%09%09timestamp%0A%09%09totalVolumeUsd%0A%09%7D%0A%0A%09tokenDailyTotals%28%0A%09%09where%3A+%7B+token%3A+%24address%2C+timestamp_gte%3A+%24from%2C+timestamp_lt%3A+%24to+%7D%0A%09%29+%7B%0A%09%09timestamp%0A%09%09totalVolumeUsd%0A%09%7D%0A%7D%0A) query. You'd need to pass 3 variables:

```json
{
  "address": "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48",
  "from": 1663459200, // yesterday
  "to": 1663545600 // today
}
```

Where address is the token address you want to check and from and to are Unix timestamps. For the Unix timestamps you can use https://www.epochconverter.com, keep in mind that to mimic how the app is getting the data you need to set 00:00hs UTC of today vs yesterday.

Regarding what we have now in prod vs what this PR changes, the values are going to be different as to implement the sorting by 24hs volume and the way the subgraph works, the data is aggregated daily, so now we are showing the volume from yesterday (before the volume was dynamic based on the moment the user entered the Explorer, so values could change if the user visited the app during the morning vs the afternoon)

**Comment by elena-zh** [comment]
Hey @ramirotw , @matextrem , thanks for fixes and explanation. 
Now I see that numbers are not so differ as on the screenshots I attached above, and look more or less aligned given the fact that they will be different on Prod and in the pr. 

Just a tiny issue I'd like to report is that it would be great to make the date format look nicer 
readable format
As an example, make it the same as we use on tooltips for other dates on Explorer
fromat

Thanks

**Review by elena-zh — APPROVED** [review]
Approved now!

**Review by ramirotw — APPROVED** [review]

**Review by alfetopito — COMMENTED** [review]
Looking great now

**Pull request merged**